### PR TITLE
CY-2814 Added read-only support for tabs

### PR DESCRIPTION
**Pull request by kubama**
Adds read-only support for `tab` attribute in page JSON files:
```
{
  "name": "Admin Dashboard",
  "widgets": [...],
  "tabs": [
    { "name": "Tab 1", "widgets": [...]},
    ...
  ]
}
```

Tabs are displayed below page level widgets:

image

Includes `PageManagement` refactoring (moving from redux to internal component state).

Cypress test will be provided in separate PR.

### CODE REVIEW

File: app/components/AddWidgetModal.js [hunk 1]
```diff
@@ -285,7 +288,7 @@ export default class AddWidgetModal extends Component {
             StageUtils.Url.url(LoaderUtils.getResourceUrl(`widgets/${widget.id}/widget.png`, widget.isCustom));
 
         return (
-            <div className={className}>
+            <div className="compactBlock">
```
Comment (qooban) [inline app/components/AddWidgetModal.js:291]: Probably we can even go further and replace `className="compactBlock"` with `style={{display: 'inline-block'}}` here and remove `compactBlock` from `styles.scss`.

File: app/components/EditModeBubble.js [hunk 2]
```diff
@@ -4,69 +4,25 @@
...
+                    <Message.Content className="alignCenter">{children}</Message.Content>
                 </Message>
             </Sidebar>
-        );
-    }
+        </>
+    );
 }
+
+EditModeBubble.propTypes = {
```
Comment (qooban) [inline app/components/EditModeBubble.js:24]: Great :+1: One component less to add propTypes.

File: app/components/PageContent.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        <>
+            {createWidgetList(page.widgets)}
+            {!_.isEmpty(page.tabs) && (
+                <Tab
+                    panes={_.map(page.tabs, (tab, tabIndex) => ({
+                        menuItem: tab.name,
+                        render: () => createWidgetList(tab.widgets, tabIndex)
+                    }))}
+                />
+            )}
```
Comment (qooban) [inline app/components/PageContent.js:28]: I assume you confirmed that limitation (tabs only below widgets and no widgets below tabs) with Ofer, if not it's just a mere formality to do so.

File: app/components/Widget.js [hunk 4]
```diff
@@ -119,10 +111,7 @@ export default class Widget extends Component {
                     <div className="widgetButtons" onMouseDown={e => e.stopPropagation()}>
                         {isEditMode && (
                             <div className="widgetEditButtons">
-                                <i
-                                    className="remove link icon small"
-                                    onClick={() => onWidgetRemoved(pageId, widget.id)}
-                                />
+                                <i className="remove link icon small" onClick={() => onWidgetRemoved(widget.id)} />
```
Comment (qooban) [inline app/components/Widget.js:114]: Let's migrate it to proper `<Icon />` component (from `semantic-ui-react`).

File: app/components/layout/Layout.js [hunk 5]
```diff
@@ -85,8 +84,19 @@ export default class Layout extends Component {
...
+                            path="/page_preview/:pageId"
+                            render={({ match }) => <PageManagement pageId={match.params.pageId} />}
+                        />
+                    )}
+                    {isUserAuthorizedForTemplateManagement && (
+                        <Route
+                            exact
+                            path="/page_edit/:pageId"
+                            render={({ match }) => <PageManagement pageId={match.params.pageId} isEditMode />}
+                        />
```
Comment (qooban) [inline app/components/layout/Layout.js:99]: Nice that you added distinction on page preview/edit on URL level. I'm just wondering if we should not have both URLs prefixed with /template_management. I'm not for or against. Just asking for your opinion?

File: app/components/templates/PageManagement.js [hunk 6]
```diff
@@ -2,124 +2,208 @@
...
+    const isWidgetMaximized = findWidget({ maximized: true });
+
+    $('body')
+        .css({ overflow: isWidgetMaximized ? 'hidden' : 'inherit' })
+        .scrollTop(0);
+
+    return (
+        <div className="main">
+            <div className="sidebarContainer">
+                <div className="ui visible left vertical sidebar menu small basic">
```
Comment (qooban) [inline app/components/templates/PageManagement.js:140]: Looks like leftover from the time where we used `semantic-ui` and not `semantic-ui-react`.
Let's migrate it to proper `semantic-ui-react` component (`<Menu />`? `<Sidebar />`?).

File: app/components/templates/PageManagement.js [hunk 7]
```diff
@@ -2,124 +2,208 @@
...
+                    <div className="pages">
+                        <div className="item link pageMenuItem">{page.name}</div>
+                        <div className="item link pageMenuItem" />
```
Comment (qooban) [inline app/components/templates/PageManagement.js:143]: `div.item.link` can probably be also migrated to `semantic-ui-react` component.

File: app/components/PageContent.js (hunk 3 above)
Comment (kubama) [reply app/components/PageContent.js:28]: This is clearly stated in story description:
> a page may contain a widget and below it a tab (or set of tabs) containing other widgets

File: app/components/layout/Layout.js (hunk 5 above)
Comment (kubama) [reply app/components/layout/Layout.js:99]: Both options seem equally good to me. I don't see any real benefit but also no issue from prefixing it with /template_management.
I would leave it as it is.

File: app/components/templates/PageManagement.js (hunk 7 above)
Comment (kubama) [reply app/components/templates/PageManagement.js]: Done

File: app/components/templates/PageManagement.js (hunk 6 above)
Comment (kubama) [reply app/components/templates/PageManagement.js]: Done

File: app/components/Widget.js (hunk 4 above)
Comment (kubama) [reply app/components/Widget.js]: Done

File: app/components/AddWidgetModal.js (hunk 1 above)
Comment (kubama) [reply app/components/AddWidgetModal.js]: Done

**Pull request merged**